### PR TITLE
Limit Network Area Diagram parallel generations

### PR DESCRIPTION
**Pull request by ayolab**
**Please check if the PR fulfills these requirements**

- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**

NO

**What kind of change does this PR introduce?**

Feature

**What is the current behavior?**

When hitting the server with many multiple requests on big networks, most probably it will run out of memory.

**What is the new behavior (if this is a feature change)?**
The server queues requests in memory and handles a fixed number in parallel to avoid OOM

**Does this PR introduce a breaking change or deprecate an API?**
- [ ] Yes
- [x] No

**If yes, please check if the following requirements are fulfilled**

- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration steps are described in the following section

**What changes might users need to make in their application due to this PR? (migration steps)**

**Other information**:

NOTE: if needed in the future, we can add some kind of budgeting (for example based on the number of substations, so that we can have unlimited requests in parallel for small networks and only a few in parallel on big networks..

### CODE REVIEW

**Review by AbdelHedhili — COMMENTED** [review]
Should we add a unit test to check that there is indeed a limit now or it's too painful / doesn't bring much ?

File: src/main/java/com/powsybl/sld/server/SingleLineDiagramController.java [hunk 1]
```diff
@@ -55,19 +55,15 @@ public class SingleLineDiagramController {
     static final String METADATA = "metadata";
     static final String ADDITIONAL_METADATA = "additionalMetadata";
 
+    // 3 maximum concurrent Network Area Diagram generations for the current configuration
+    private final ExecutorService executorService = Executors.newFixedThreadPool(3);
```
Comment (AbdelHedhili) [inline src/main/java/com/powsybl/sld/server/SingleLineDiagramController.java:59]: Shouldn't we externalize the number of ThreadPool in a conf ?

Comment (ayolab) [reply src/main/java/com/powsybl/sld/server/SingleLineDiagramController.java]: Done

File: src/main/java/com/powsybl/sld/server/SingleLineDiagramController.java [hunk 2]
```diff
@@ -265,14 +277,21 @@ public ResponseEntity<Collection<String>> getAvailableSvgComponentLibraries() {
...
+                SvgAndMetadata svgAndMetadata = networkAreaDiagramService.generateNetworkAreaDiagramSvg(networkUuid, variantId, voltageLevelsIds, depth);
+                String svg = svgAndMetadata.getSvg();
+                Object additionalMetadata = svgAndMetadata.getAdditionalMetadata();
+                return OBJECT_MAPPER.writeValueAsString(
+                        OBJECT_MAPPER.createObjectNode()
+                                .put(SVG_TAG, svg)
+                                .putPOJO(ADDITIONAL_METADATA, additionalMetadata));
+            } catch (JsonProcessingException e) {
+                throw new PowsyblException("Failed to parse JSON response", e);
+            }
```
Comment (AbdelHedhili) [inline src/main/java/com/powsybl/sld/server/SingleLineDiagramController.java:293]: can you add a test where you go in this catch to reach the coverage threshold

**Review by AbdelHedhili — CHANGES_REQUESTED** [review]

**Comment by jonenst** [comment]
missing predestroy shutdown of executorservice

File: src/main/java/com/powsybl/sld/server/SingleLineDiagramController.java (hunk 2 above)
Comment (jonenst) [reply src/main/java/com/powsybl/sld/server/SingleLineDiagramController.java:293]: looks like its unreachable code. I don't think we care about the coverage

**Review by AbdelHedhili — APPROVED** [review]

**Review by jonenst — APPROVED** [review]

**Pull request merged**